solidity/test/TestBeaconRewards.js

```diff
@@ -0,0 +1,161 @@
+const { initContracts } = require('./helpers/initContracts')
+const { accounts, contract, web3 } = require("@openzeppelin/test-environment")
+const { createSnapshot, restoreSnapshot } = require("./helpers/snapshot.js")
+const { expectRevert, time } = require("@openzeppelin/test-helpers")
+const crypto = require("crypto")
+const stakeDelegate = require('./helpers/stakeDelegate')
+
+const BN = web3.utils.BN
+const chai = require('chai')
+chai.use(require('bn-chai')(BN))
+const expect = chai.expect
+const assert = chai.assert
+
+describe('BeaconRewards', () => {
+
+    let token, stakingContract, operatorContract, serviceContract, rewards,
+        groupSize,
+        group1, group2, group3,
+        owner = accounts[0],
+        requestor = accounts[1],
+        operator1 = accounts[2],
+        operator2 = accounts[3],
+        operator3 = accounts[4],
+        beneficiary1 = accounts[5],
+        beneficiary2 = accounts[6],
+        beneficiary3 = accounts[7]
+
+    before(async () => {
+        let contracts = await initContracts(
+            contract.fromArtifact('TokenStaking'),
+            contract.fromArtifact('KeepRandomBeaconService'),
+            contract.fromArtifact('KeepRandomBeaconServiceImplV1'),
+            contract.fromArtifact('KeepRandomBeaconOperatorBeaconRewardsStub')
+        )
+        const termLength = 10
+        const totalRewards = 6000
+        const minimumIntervalKeeps = 2
```

Comment:
would love to see a test playing with 0 <= minimumIntervalKeeps < 2

Reply:
With the new allocation logic (using `Math.max(minimumIntervalKeeps, keepCount)`) the only situation that's really different is when both are zero, which used to be a divide by zero revert from SafeMath but now allocates 0 correctly.